### PR TITLE
Update the test "test_rolling_terminate_and_recovery" for MS and non-MS clusters

### PR DESCRIPTION
**Pull request by yitzhak12**
Update the test "test_rolling_terminate_and_recovery" for MS and non-MS clusters:
- Generate nodes for a provider cluster using the function 'generate_nodes_for_provider_worker_node_tests' if it's a provider cluster. 
- Wait for the new worker node to be ready
- Label the new worker node in case of a non-MS cluster

### CODE REVIEW

**Review by DanielOsypenko — APPROVED** [review]

**Comment by yitzhak12** [comment]
The pr validation passed with provider and consumer clusters: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/1475/console.

File: tests/manage/z_cluster/nodes/test_rolling_terminate_and_recovery.py [hunk 1]
```diff
@@ -74,17 +85,36 @@ def finalizer():
 
     def rolling_terminate_and_recovery_of_ocs_worker_nodes(self, nodes):
```
Comment (fbalak) [inline tests/manage/z_cluster/nodes/test_rolling_terminate_and_recovery.py:86]: Please add a docstring to the function. Please add an explanation about what happens with and without ms.

Comment (yitzhak12) [reply tests/manage/z_cluster/nodes/test_rolling_terminate_and_recovery.py:86]: Okay, that will do.

File: tests/manage/z_cluster/nodes/test_rolling_terminate_and_recovery.py [hunk 2]
```diff
@@ -74,17 +85,36 @@ def finalizer():
...
         # Get OCS worker node objects
-        ocs_node_objs = get_ocs_nodes()
+        if is_ms_provider_cluster():
+            ocs_node_objs = generate_nodes_for_provider_worker_node_tests()
+        else:
+            # If it's not a provider cluster, test rolling terminate two ocs worker nodes will suffice
+            ocs_node_objs = random.choices(get_ocs_nodes(), k=2)
 
-        # Start rolling shutdown and recovery of OCS worker nodes
+        # Start rolling terminate and recovery of OCS worker nodes
```
[the remark this answers is not in the document]
Comment (yitzhak12) [reply tests/manage/z_cluster/nodes/test_rolling_terminate_and_recovery.py]: Fixed

File: tests/manage/z_cluster/nodes/test_rolling_terminate_and_recovery.py [hunk 3]
```diff
@@ -74,17 +85,36 @@ def finalizer():
...
             nodes.terminate_nodes(nodes=[node_obj], wait=True)
             log.info(f"Successfully terminated the node: {node_obj.name}")
+            if is_managed_service_cluster():
+                new_ocs_node = wait_for_new_worker_node_ipi(machineset, old_wnodes)
+            else:
+                expected_ready_rc = old_ready_rc - 1
+                wait_for_ready_replica_count_to_reach_expected_value(
+                    machineset, expected_ready_rc
+                )
+                # Change the current replica count to the expected ready replica count
```
[the remark this answers is not in the document]
Comment (yitzhak12) [reply tests/manage/z_cluster/nodes/test_rolling_terminate_and_recovery.py]: Fixed

**Review by fbalak — APPROVED** [review]

**Comment by yitzhak12** [comment]
The pr validation passed with provider and consumer clusters: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/1481/.

**Review by DanielOsypenko — APPROVED** [review]

**Pull request merged**